### ISSUE TITLE
Upscale to target resolution for seamless tileable textures

### ISSUE THREAD

**Issue by rexelbartolome** [chaiNNer-org/chaiNNer#1505]
So this is kind of two problems that have have already been mentioned somewhere here #830 and #1498

Say I use a 4x model and I'm upscaling a seamless 512x512 image. I want my target resolution to be 4096x4096. One way of going about it is

![chainner_byx4sDPaum](https://user-images.githubusercontent.com/88552647/214634801-f8c2d831-1b0a-4580-a10d-cb24e4572579.png)

After testing, it made a seam on the edges of the texture when I tile it. Is there something wrong with my math or how I understand the border thing?
![krita_8ypusQeQYb](https://user-images.githubusercontent.com/88552647/214635077-b6244d6e-15dc-48fe-a06c-fd4acb718e8e.png)

Other questions:

I feel like upscaling twice using a 4x model then downscaling it to almost half its size is overkill and inefficient, is that just how it is or am I doing something wrong? Also, is there a standard to how people upscale their textures? Couldn't find much resources on youtube.

If I say change from a 4x model to a 2x model, is there a preset that i can follow to find the optimal times I need to upscale? Or is there a recursive upscale node then somehow hook that up to a math node comparing the resolution to the target resolution?

**Comment by joeyballentine** [comment]
You cropped off too much. You should have ended up with 16 x 4 x 4, or 256 but somehow you got 272

**Comment by RunDevelopment** [comment]
>  Is there something wrong with my math

Yes, you are removing too much padding. Create border adds 16px of padding, so after upscaling it 4x and another 4x, you have to remove 16px\*4\*4=256px of padding.

The underlying problem is that the second math node multiplies the upscale dimensions which include padding with the same factor as the first math node. This just doesn't work.

**Comment by rexelbartolome** [comment]
Hmmm yea okay I see it now :) then it should be something like this 
![image](https://user-images.githubusercontent.com/88552647/214644347-76175db6-066f-4739-8bb3-ba4aba544d8c.png)

seams are gone now too, thanks :D
![krita_0Kmrp4tDX3](https://user-images.githubusercontent.com/88552647/214644897-e276e35c-dc6d-4732-a216-8d7e8a7cc90a.png)

but how bout my other questions? Is this just how people do them or is there a better way? I'm fairly new to upscaling so I don't really know...

**Comment by theflyingzamboni** [comment]
No, that's how people do it. Model scales are invariant, so the only way to hit specific resolution targets if they aren't `model scale x original resolution` is to downscale at some stage of the process.

**Comment by rexelbartolome** [comment]
damn, alright then
